guppy/src/graph/cargo/cargo_api.rs

```diff
@@ -254,9 +254,30 @@ impl<'g> CargoSet<'g> {
         initials: FeatureSet<'g>,
         features_only: FeatureSet<'g>,
         opts: &CargoOptions<'_>,
+    ) -> Result<Self, Error> {
+        Self::new_internal(initials, features_only, None, opts)
+    }
+
+    /// Like `Cargo.new`, but allows inspecting and influencing dependency traversal with a custom
+    /// `resolver`.
```

Comment:
I don't think this is quite the right description -- it seems to me that this acts as effectively an additional filter on `PackageLink`? It's similar to `omitted_packages`, but on graph edges rather than nodes.

If so, I think the naming and documentation should be geared towards that. (And I think it would also then make sense to accept a `PackageResolver` as a filter. In the future we can add an even richer API.)

This would also benefit from an example use case. I'd recommend adding something to the `examples` directory.